firewood/src/merkle/node.rs

```diff
@@ -402,6 +416,113 @@ impl<T> PartialEq for EncodedNode<T> {
     }
 }
 
+#[derive(Debug)]
+struct PathWithBitsPrefix(Path);
+
+impl<'de> Deserialize<'de> for PathWithBitsPrefix {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        // TODO danlaine: This is slightly abusing the serde API.
+        // We don't know the number of bytes in the tuple until we've read the first u64.
+        // We should implement a custom serializer/deserializer for EncodedNode.
+        deserializer.deserialize_tuple(usize::MAX, TupleVisitor)
+    }
+}
+
+struct TupleVisitor;
```

Comment:
It's actually common to define the `Visitor` directly in the `deserialize` implementation if it's not meant to be used anywhere else. Also, this name implies that the visitor is more generic than it actually is.